components/raftstore/src/coprocessor/dispatcher.rs

```diff
@@ -665,6 +665,16 @@ impl<E: KvEngine> CoprocessorHost<E> {
         }
     }
 
+    pub fn on_update_safe_ts(&self, region_id: u64, self_safe_ts: u64, leader_safe_ts: u64) {
```

Comment:
Not sure which observer is better, since query observer is used when we observe applying/executing of raft write commands. However, other raft commands can also advance `safe_ts`.
Is region change observer better? /cc @BusyJay

Reply:
I create a new observer.